src/views/Playgrounds/PlaygroundForm.tsx

```diff
@@ -0,0 +1,120 @@
+import { useForm, Controller } from 'react-hook-form'
+import {
+  Button as _Button,
+  Text,
+  FormField,
+  TextField,
+  Select,
+  Checkbox,
+  Textarea,
+  HeaderTextField,
+} from '@/shared/components'
+import { textFieldValidation, requiredValidation } from './formValidationOptions'
+import { SelectedItem } from '@/shared/components/Select'
+import styled from '@emotion/styled'
+
+const items: SelectedItem[] = [
+  { name: 'Public (Anyone can see this video', value: 'public' },
+  { name: 'Private', value: 'private' },
+]
+
+type Inputs = {
+  title: string
+  videoSelect: string
+  header: string
+  check: string
+  textarea: string
+}
+
+const PlaygroundForm = () => {
+  const { register, handleSubmit, control, setValue, errors } = useForm<Inputs>({
+    shouldFocusError: false,
+  })
+  const [selectedItem, setSelectedItem] = useState<SelectedItem | null>(null)
+  const [checkboxValue, setCheckboxValue] = useState(false)
+
+  const onSubmit = handleSubmit((data) => console.log(data))
+
+  return (
+    <>
+      <form onSubmit={onSubmit}>
+        <HeaderTextField
+          name="header"
+          ref={register(textFieldValidation('Channel name', 3, 20))}
+          value="Lorem ipsum"
+          error={!!errors.header}
+          helperText={errors.header?.message}
+        />
+
+        <FormField title="Title" description="Lorem ipsum dolor sit amet">
+          <TextField
+            name="title"
+            label="Title"
+            ref={register(textFieldValidation('Title', 3, 20))}
+            error={!!errors.title}
+            helperText={errors.title?.message}
+          />
+        </FormField>
+
+        <FormField title="Video Visibility">
+          <Controller
+            name="videoSelect"
+            control={control}
+            defaultValue={selectedItem}
+            rules={requiredValidation('Video Visibility')}
+            render={() => (
+              <Select
+                items={items}
+                onChange={({ selectedItem }) => {
+                  setValue('videoSelect', selectedItem)
+                  selectedItem && setSelectedItem(selectedItem)
+                }}
+                value={selectedItem}
+                error={!!errors.videoSelect && !selectedItem}
+                helperText={errors.videoSelect?.message}
+              />
+            )}
+          />
+        </FormField>
+
+        <FormField title="Marketing" description="Lorem ipsum dolor sit amet.">
+          <StyledCheckboxContainer>
+            <Checkbox
+              name="check"
+              ref={register(requiredValidation('Checkbox'))}
+              value={checkboxValue}
+              onChange={setCheckboxValue}
+              error={!!errors.check}
+            />
+            <Text>My video features a paid promotion material</Text>
+          </StyledCheckboxContainer>
+        </FormField>
+        <FormField title="Description">
+          <Textarea
+            name="textarea"
```

Comment:
```suggestion
            name="textarea"
            maxLength={20}
```

Small thing, but you can pass also `maxLength` to `TextArea`. ;)

![image](https://user-images.githubusercontent.com/51168865/108858446-732d1c80-75ec-11eb-943f-a200ba62af9a.png)

Reply:
Yup, was aware of that